account-kit/smart-contracts/src/ma-v2/account/semiModularAccountV2.ts

```diff
@@ -20,18 +21,53 @@ import {
   type Hex,
   type Transport,
   hexToBigInt,
+  zeroAddress,
 } from "viem";
 import { accountFactoryAbi } from "../abis/accountFactoryAbi.js";
 import { getDefaultMAV2FactoryAddress } from "../utils.js";
 import { standardExecutor } from "../../msca/account/standardExecutor.js";
 import { singleSignerMessageSigner } from "../modules/single-signer-validation/signer.js";
 import { InvalidEntityIdError, InvalidNonceKeyError } from "@aa-sdk/core";
+import { modularAccountAbi } from "../abis/modularAccountAbi.js";
+import { serializeModuleEntity } from "../actions/common/utils.js";
+
+const executeUserOpSelector: Hex = "0x8DD7712F";
 
 export const DEFAULT_OWNER_ENTITY_ID = 0;
 
+export type CalldataEncoder = {
+  encodeCallData: (callData: Hex) => Promise<Hex>;
+};
+
+export type ExecutionDataView = {
+  module: Address;
+  skipRuntimeValidation: boolean;
+  allowGlobalValidation: boolean;
+  executionHooks: readonly Hex[];
```

Comment:
Is there are reason some fields are in this type and the next are `readonly` and others are not?

As personal style, I don't use `readonly` on data structs. IMO it doesn't meaningfully prevent mutations if you rely on that for correctness, and just makes things inconvenient otherwise. For example, the following typechecks:
```ts
type Circle = { readonly r: number };

function useCircle(circle: Circle) {
  // Do something with circle that would break
  // if it later changes.
}

const circle = { r: 10 };
useCircle(circle);
circle.r = 11;
```

Reply:
without the readonly I get a type error, it seems like the return data from the view call is readonly for all Hex types

Reply:
is there a way to remove the readonlys?